### PR TITLE
stable

### PR DESCRIPTION
**Pull request by gandesc**

### CODE REVIEW

File: src/database/migrations/2017_12_07_151000_create_addresses_table.php [hunk 1]
```diff
@@ -16,28 +16,22 @@ public function up()
...
-            $table->string('sub_administrative_area')->nullable();
-            $table->string('city');
-            $table->string('administrative_area')->nullable();
+            $table->integer('locality_id')->unsigned()->index()->nullable();
+            $table->foreign('locality_id')->references('id')->on('localities');
 
+            $table->string('city')->nullable();
+            $table->string('street');
+            $table->string('additional')->nullable();
             $table->string('postal_area')->nullable();
```
Comment (aocneanu) [inline src/database/migrations/2017_12_07_151000_create_addresses_table.php:28]: this should be postcode

File: src/database/factories/AddressFactory.php [hunk 2]
```diff
@@ -4,12 +4,22 @@
...
+    return [
+        'addressable_id' => $faker->randomNumber(5),
+        'addressable_type' => $faker->word,
+        'country_id' => $country->id,
+        'region_id' => optional($country->regions()->first())->id,
+        'locality_id' => optional($country->localities()->first())->id,
+        'city' => $faker->city,
+        'street' => $faker->streetAddress,
+        'additional' => null,
+        'postcode' => $faker->randomNumber(5),
```
Comment (aocneanu) [inline src/database/factories/AddressFactory.php:19]: $faker->postcode maybe?

Comment (gandesc) [reply src/database/factories/AddressFactory.php]: done

File: src/App/Models/Locality.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+class Locality extends Model implements Activatable
+{
+    use ActiveState;
+
+    protected $fillable = [
+        'region_id', 'name', 'township', 'siruta', 'lat', 'long', 'is_active',
+    ];
+
+    protected $appends = ['label'];
```
Comment (aocneanu) [inline src/App/Models/Locality.php:17]: do we still want this?

**Pull request merged**